src/sessions/p2p_session.rs

```diff
@@ -842,4 +862,49 @@ impl<T: Config> P2PSession<T> {
             self.event_queue.pop_front();
         }
     }
+
+    fn compare_local_checksums_against_peers(&mut self) {
+        let current = self.current_frame();
+        if current % CHECKSUM_REPORT_INTERVAL == 0 && current > self.max_prediction as i32 {
```

Comment:
What is the current > max_prediction for?

Reply:
this guard appeard during the implementation of the system. i just tested it and its now needed anymore.